docs(api): add `defineNuxtComponent` utils (#7618)
nuxt/framework: docs/content/3.api/3.utils/define-nuxt-component.md

# `defineNuxtComponent`

`defineNuxtComponent()` is a helper function for defining type safe Vue components within a Nuxt application. `defineNuxtComponent()` is a wrapper around [defineComponent()](https://vuejs.org/api/general.html#definecomponent) function that supports inferring the props passed to `setup()` function when we use Composition API without `<script setup>`.

> `defineNuxtComponent()` also takes care of resolving the promise returned by `useAsyncData()` in Vue components found at `components/` and `pages/` directory or in the `~/app.vue` component found at the root of your Nuxt application.

Reviewer: The main reason you would use `defineNuxtComponent` is to support the Options API version of asyncData, so that might be worth mentioning in the example below.

Reviewer 2: Checking current implementation, indeed it also gives type support but same as `defineComponent` with only addition of `asyncData` support.

Any Vue 3 components written in Nuxt application without `<script setup lang=“ts”>` can use `defineNuxtComponent()` helper function to leverage the benefits of TypeScript.

## Example

```vue [pages/index.vue]
  <script lang="ts">

>     import { defineComponent } from 'vue'

Reviewer: ```suggestion
```

Reviewer 3: Can we keep one import statement for users who want to explicitly add the import?

    export default defineNuxtComponent({
      async setup(props) {

>         const { data } = await useAsyncData('count', () => Promise.resolve({data: { name: 'Hello world!'}}))

Reviewer: ```suggestion
        const { data } = await useAsyncData(() => Promise.resolve('Hello world!'))
```

# `defineNuxtComponent`

`defineNuxtComponent()` is a helper function for defining type safe Vue components using options API similar to [defineComponent()](https://vuejs.org/api/general.html#definecomponent). `defineNuxtComponent()` wrapper also adds support for `asyncData` component option.

::alert{type=warning}
Options API support for `asyncData` may well change before the stable release of Nuxt 3.
::

::Alert
Using `<script setup lang="ts">` is the recommended way of declaring Vue components in Nuxt 3.
::

:ReadMore{link=/getting-started/data-fetching#options-api-support}

## `asyncData()`

> If you choose to use options API to define components in your Nuxt 3, you can use `asyncData()` utility:

Reviewer: ```suggestion
If you choose not to use `setup()` in your app, you can use the `asyncData()` method within your component definition:
```

```vue [pages/index.vue]
<script lang="ts">
export default defineNuxtComponent({

>   asyncData() {

Reviewer: ```suggestion
  async asyncData() {
```
